InventoryBundleProduct/Model/GetBundleProductStockStatus.php

```diff
@@ -0,0 +1,86 @@
+<?php
+/**
+ * Copyright © Magento, Inc. All rights reserved.
+ * See COPYING.txt for license details.
+ */
+declare(strict_types=1);
+
+namespace Magento\InventoryBundleProduct\Model;
+
+use Magento\Bundle\Api\Data\OptionInterface;
+use Magento\Bundle\Model\OptionRepository;
+use Magento\Catalog\Api\Data\ProductInterface;
+use Magento\InventorySalesApi\Api\IsProductSalableInterface;
+
+/**
+ * Get bundle product stock status service.
+ */
+class GetBundleProductStockStatus
+{
+    /**
+     * @var OptionRepository
+     */
+    private $optionRepository;
+
+    /**
+     * @var GetProductSelection
+     */
+    private $getProductSelection;
+
+    /**
+     * @var IsProductSalableInterface
+     */
+    private $isProductSalable;
+
+    /**
+     * GetBundleProductStockStatus constructor
+     *
+     * @param OptionRepository $optionRepository
+     * @param GetProductSelection $getProductSelection
+     * @param IsProductSalableInterface $isProductSalable
+     */
+    public function __construct(
+        OptionRepository $optionRepository,
+        GetProductSelection $getProductSelection,
+        IsProductSalableInterface $isProductSalable
+    ) {
+        $this->optionRepository = $optionRepository;
+        $this->getProductSelection = $getProductSelection;
+        $this->isProductSalable = $isProductSalable;
+    }
+
+    /**
+     * Provides bundle product stock status.
+     *
+     * @param ProductInterface $product
+     * @param OptionInterface[] $bundleOptions
+     * @param int $stockId
+     *
+     * @return bool
+     */
+    public function execute(ProductInterface $product, array $bundleOptions, int $stockId): bool
+    {
+        $isSalable = false;
+        foreach ($bundleOptions as $option) {
+            $hasSalable = false;
+            $bundleSelections = $this->getProductSelection->execute($product, $option);
```

Comment:
Performance issue here?

Reply:
@Stepa4man, thanks for remark. Currently we do not have salable check for multiple products, so it will be fixed in next PR, which will introduce bulk "is salable" verification.